Odysee/Controllers/Content/CommentsViewController.swift

```diff
@@ -369,79 +346,59 @@ class CommentsViewController: UIViewController, UITableViewDelegate, UITableView
-                        message
-                    ))
-                }
-
-                do {
-                    let commentData = try JSONSerialization.data(
-                        withJSONObject: result,
-                        options: [.prettyPrinted, .sortedKeys]
-                    )
-                    let comment = try JSONDecoder().decode(Comment.self, from: commentData)
-                    if let currentReplyToComment = self.currentReplyToComment {
-                        if currentReplyToComment.repliesLoaded ?? false {
-                            let parentIndex = self.indexForComment(currentReplyToComment)
-                            if parentIndex > -1 {
-                                self.comments.insert(comment, at: parentIndex + 1)
-                            }
-                        } else {
-                            DispatchQueue.main.async {
-                                self.loadReplies(currentReplyToComment)
-                            }
+            method: Lbry.Methods.channelSign,
+            params: .init(
+                channelId: channel.claimId!,
+                hexdata: Helper.strToHex(commentInput.text!)
+            )
+        )
+        .flatMap { channelSignResult in
+            return Lbry.commentApiCall(
+                method: Lbry.CommentMethods.create,
+                params: .init(
+                    claimId: self.claimId!,
+                    channelId: channel.claimId!,
+                    signature: channelSignResult.signature,
+                    signingTs: channelSignResult.signingTs,
+                    comment: self.commentInput.text!,
+                    parentId: self.currentReplyToComment?.commentId
+                )
+            )
+        }
+        .subscribeResult { result in
+            switch result {
+            case let .failure(error):
+                self.showError(error: error)
+            case let .success(comment):
+                if let currentReplyToComment = self.currentReplyToComment {
+                    if currentReplyToComment.repliesLoaded ?? false {
+                        let parentIndex = self.indexForComment(currentReplyToComment)
+                        if parentIndex > -1 {
+                            self.comments.insert(comment, at: parentIndex + 1)
                         }
                     } else {
-                        self.comments.insert(comment, at: 0)
+                        self.loadReplies(currentReplyToComment)
                     }
-                    self.loadCommentReactions(commentIds: [comment.claimId!])
-                    self.resolveCommentAuthors(urls: [comment.channelUrl!])
-                } catch {
-                    self.showError(error: error)
+                } else {
+                    self.comments.insert(comment, at: 0)
                 }
+                // TODO: Reactions not loaded here as comment might not be available?
```

Comment:
@akinwale can you please take a look at this?

Reply:
Still investigating this. I'll tackle the issue with a new PR.